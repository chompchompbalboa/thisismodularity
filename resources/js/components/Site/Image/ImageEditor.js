//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import { number, oneOf, shape, string } from 'prop-types'

import AppDashboardContentContainerSection from '../../App/AppDashboard/AppDashboardContentContainerSection/AppDashboardContentContainerSection'
import EditImage from '../../lib/EditImage/EditImage'
import EditPadding from '../../lib/EditPadding/EditPadding'
import EditWidthHeight from '../../lib/EditWidthHeight/EditWidthHeight'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
const ImageEditor = ({ activeDevice, activeModule: { data }, activeModuleIndex, activePageIndex }) => (
  <AppDashboardContentContainerSection>
    <EditImage
      activeDevice={activeDevice}
      activeModuleIndex={activeModuleIndex}
      activePageIndex={activePageIndex}
      image={data.src}
      imageKey="src"/>
    <EditWidthHeight
      activeDevice={activeDevice}
      activeModuleIndex={activeModuleIndex}
      activePageIndex={activePageIndex}
      height={data.height}
      width={data.width}/>
    <EditPadding
      activeDevice={activeDevice}
      activeModuleIndex={activeModuleIndex}
      activePageIndex={activePageIndex}
      padding={data.padding}/>
  </AppDashboardContentContainerSection>
)

//-----------------------------------------------------------------------------
// Props
//-----------------------------------------------------------------------------
ImageEditor.propTypes = {
  activeDevice: oneOf([
    "MOBILE",
    "TABLET",
    "DESKTOP"
  ]),
  activeModule: shape({
    data: shape({
      text: string,
      fontFamily: string
    })
  }),
  activeModuleIndex: number,
  activePageIndex: number
}

export default ImageEditor