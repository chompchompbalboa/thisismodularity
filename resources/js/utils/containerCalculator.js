//-----------------------------------------------------------------------------
// Container Dimensions
//-----------------------------------------------------------------------------
export const containerDimensions = {
  SITE: {
    width: window.innerWidth,
    height: window.innerHeight
  },
  MOBILE: {
    width: (((window.innerHeight * 0.75) * 9) / 16),
    height: (window.innerHeight * 0.75)
  },
  TABLET: {
    width: (((window.innerHeight * 0.93) * 3) / 4),
    height: (window.innerHeight * 0.93)
  },
  DESKTOP: {
    width: window.innerWidth * 0.96,
    height: window.innerHeight
  }
}

//-----------------------------------------------------------------------------
// Calculator
//-----------------------------------------------------------------------------
export const calculator = {
  backgroundImage: (activeDevice, backgroundImage) => {
    return "url(" + backgroundImage + ")"
  },
  fontSize: (activeDevice, fontSize) => {
    return (
      ((containerDimensions[activeDevice].height * (fontSize[activeDevice] / 100)) / 5) + "px "
    )
  },
  height: (activeDevice, height) => {
    if(height[activeDevice].auto) {
      return "auto"
    }
    return (containerDimensions[activeDevice].height * (height[activeDevice].value / 100)) + "px"
  },
  padding: (activeDevice, padding) => {
    return (
      (containerDimensions[activeDevice].height * (padding[activeDevice].TOP / 100)) + "px " +
      (containerDimensions[activeDevice].width * (padding[activeDevice].RIGHT / 100)) + "px " +
      (containerDimensions[activeDevice].height * (padding[activeDevice].BOTTOM / 100)) + "px " +
      (containerDimensions[activeDevice].width * (padding[activeDevice].LEFT / 100)) + "px"
    )
  },
  width: (activeDevice, width) => {
    if(width[activeDevice].auto) {
      return "auto"
    }
    return (containerDimensions[activeDevice].width * (width[activeDevice].value / 100)) + "px"
  },
}


//-----------------------------------------------------------------------------
// Get Device Style
//-----------------------------------------------------------------------------
export const getDeviceStyle = ( activeDevice, isApp ) => {
  let deviceStyle = {};
  let windowWidth = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;

  if (!isApp || windowWidth < 1024) {
    deviceStyle = {
      position: "fixed",
      top: "0vh",
      left: "0vw",
      width: "100vw",
      height: "100vh",
      overflowY: "scroll",
      boxShadow: "none"
    }
  }
  else {
      switch (activeDevice) {
          case "DESKTOP":
              deviceStyle = {
                  position: "absolute",
                  top: "0vh",
                  left: "4vw",
                  width: "96vw",
                  height: "100vh",
                  overflowY: "scroll",
                  overflowX: "hidden",
                  backgroundColor: "white",
                  transition: "all 0.5s",
                  boxShadow: "none"
              }
          break;
          case "TABLET":
              deviceStyle = {
                  position: "absolute",
                  top: "3.5vh",
                  left: "calc(27.5vw + ((72.5vw - ((93vh * 3)/4)) / 2))",
                  width: "calc((93vh * 3)/4)",
                  height: "93vh",
                  overflowY: "scroll",
                  overflowX: "hidden",
                  backgroundColor: "white",
                  transition: "all 0.5s",
                  boxShadow: "2px 2px 5px black"
              }
          break;
          default:
              deviceStyle = {
                  position: "absolute",
                  top: "12.5vh",
                  left: "calc(27.5vw + ((72.5vw - ((75vh * 9)/16)) / 2))",
                  width: "calc((75vh * 9)/16)",
                  height: "75vh",
                  overflowY: "scroll",
                  overflowX: "hidden",
                  backgroundColor: "white",
                  transition: "all 0.5s",
                  boxShadow: "2px 2px 5px black"
              }
          break;
      }
  }
  return deviceStyle;
}