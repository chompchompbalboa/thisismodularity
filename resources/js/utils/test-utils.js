//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import { render } from 'react-testing-library'

//-----------------------------------------------------------------------------
// Custom Render
//-----------------------------------------------------------------------------
const customRender = (node, options) => {
  return render(
    node,
    options,
  )
}

//-----------------------------------------------------------------------------
// Exports
//-----------------------------------------------------------------------------
export * from 'react-testing-library'
export { customRender as render }