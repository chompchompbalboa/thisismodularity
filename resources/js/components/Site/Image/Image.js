//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React, { Component } from 'react'
import { bool, number, object, oneOf, shape, string } from 'prop-types'
import styled from 'styled-components'

import { calculator } from '../../../utils/containerCalculator'

import module from '../module/module'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
class Image extends Component {
  render() {
    const {
      activeDevice,
      data: {
        src,
        width,
        height,
        padding
      },
      moduleRef
    } = this.props

    return (
      <Container
        data-testid="container"
        innerRef={moduleRef}
        activeDevice={activeDevice}
        styledProps={{
          width: width,
          height: height,
          padding: padding
        }}>
        <StyledImage
          data-testid="image"
          activeDevice={activeDevice}
          styledProps={{
            src: src
          }}/>
      </Container>
    )
  }
}

//-----------------------------------------------------------------------------
// Props
//-----------------------------------------------------------------------------
Image.propTypes = {
  activeDevice: oneOf([
    "MOBILE",
    "TABLET",
    "DESKTOP"
  ]),
  data: shape({
    src: string,
    height: shape({
      MOBILE: shape({auto: bool, value: number}),
      TABLET: shape({auto: bool, value: number}),
      DESKTOP: shape({auto: bool, value: number}),
    }),
    width: shape({
      MOBILE: shape({auto: bool, value: number}),
      TABLET: shape({auto: bool, value: number}),
      DESKTOP: shape({auto: bool, value: number}),
    }),
    padding: shape({
      MOBILE: shape({TOP: number, RIGHT: number, BOTTOM: number, LEFT: number}),
      TABLET: shape({TOP: number, RIGHT: number, BOTTOM: number, LEFT: number}),
      DESKTOP: shape({TOP: number, RIGHT: number, BOTTOM: number, LEFT: number}),
    })
  }),
  moduleRef: object
}

Image.defaultProps = {
  activeDevice: "MOBILE",
  data: {
    src: "/img/stock/1.jpg",
    height: {
      MOBILE: {auto: false, value: 50},
      TABLET: {auto: false, value: 50},
      DESKTOP: {auto: false, value: 50},
    },
    width: {
      MOBILE: {auto: false, value: 100},
      TABLET: {auto: false, value: 100},
      DESKTOP: {auto: false, value: 100},
    },
    padding: {
      MOBILE: {TOP: 5, RIGHT: 5, BOTTOM: 5, LEFT: 5},
      TABLET: {TOP: 5, RIGHT: 5, BOTTOM: 5, LEFT: 5},
      DESKTOP: {TOP: 5, RIGHT: 5, BOTTOM: 5, LEFT: 5},
    }
  }
}

//-----------------------------------------------------------------------------
// Styled Components
//-----------------------------------------------------------------------------
const Container = styled.div`
  width: ${props => calculator.width(props.activeDevice, props.styledProps.width)};
  height: ${props => calculator.height(props.activeDevice, props.styledProps.height)};
  padding: ${props => calculator.padding(props.activeDevice, props.styledProps.padding)};
`

const StyledImage = styled.div`
  width: 100%;
  height: 100%;
  background-position: center center;
  background-repeat: no-repeat;
  background-size: cover;
  background-image: ${props => calculator.backgroundImage(props.activeDevice, props.styledProps.src)};
  transition: all 0.5s;
`

export default module(Image)