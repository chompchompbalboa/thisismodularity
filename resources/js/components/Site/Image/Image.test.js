//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import 'jest-styled-components'
import { cleanup, render } from '../../../utils/test-utils'

import Image from './Image'

//-----------------------------------------------------------------------------
// Setup
//-----------------------------------------------------------------------------
afterEach(cleanup)
const component = () => {
  const renderedComponent = render(<Image />)
  return {
    ...renderedComponent
  }
}
//-----------------------------------------------------------------------------
// Tests
//-----------------------------------------------------------------------------
describe('Image', () => {
  it('renders correctly using default props', () => {
    const { getByTestId } = component()
    const container = getByTestId('container')
    const image = getByTestId('image')

    expect(container).toMatchSnapshot()
    expect(container).toHaveStyleRule('width', '324px')
    expect(container).toHaveStyleRule('height', '288px')
    expect(container).toHaveStyleRule('padding', '28.8px 16.2px 28.8px 16.2px')

    expect(image).toHaveStyleRule('background-image', 'url(/img/stock/1.jpg)')
  })
})