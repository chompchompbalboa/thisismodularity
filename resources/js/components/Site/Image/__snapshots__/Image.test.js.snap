// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly using default props 1`] = `
.c0 {
  width: 324px;
  height: 288px;
  padding: 28.8px 16.2px 28.8px 16.2px;
}

.c1 {
  width: 100%;
  height: 100%;
  background-position: center center;
  background-repeat: no-repeat;
  background-size: cover;
  background-image: url(/img/stock/1.jpg);
  -webkit-transition: all 0.5s;
  transition: all 0.5s;
}

<div
  class="c0"
  data-testid="container"
>
  <div
    class="c1"
    data-testid="image"
  />
</div>
`;
